bin/i18n/sync-codeorg-in.rb

```diff
@@ -45,6 +47,40 @@ def redact_block_content
   redact_translated_data('dashboard/blocks.yml', 'blockfield')
 end
 
+def get_dsl_i18n_strings(level)
+  return {} unless level.is_a?(DSLDefined)
+  text = level.dsl_text
+  return {} unless text
+  _, i18n = level.class.dsl_class.parse(text, '')
+  i18n
+end
+
+def localize_dsl_content
+  # Rewrite autogenerated 'dsls.en.yml' i18n file with new master-copy English strings
+  yml_file = Rails.root.join("config/locales/dsls.en.yml")
+  i18n_strings = {}
+
+  # We have to run this specifically from the Rails directory because
+  # get_dsl_i18n_strings relies on level.dsl_text which relies on
+  # level.filename which relies on running a shell command
+  Dir.chdir(Rails.root) do
+    Script.all.each do |script|
+      next unless ScriptConstants.i18n? script.name
+
+      script.levels.each do |level|
```

Comment:
So this runs through all levels, not just DSL ones right? Then the function above only runs on DSL levels?

I'm guessing this is where you're referring to refactoring localize_level_content then :)

Reply:
Precisely!